packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2752,6 +2780,28 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         }
     }
 
+    private verifyCanSubmitOps() {
+        if (!this.enableOpReentryCheck) {
+            return;
+        }
+
+        if (this.ensureNoDataModelChangesCalls > 0) {
```

Comment:
I think it's better to have

if (this.enableOpReentryCheck && this.ensureNoDataModelChangesCalls > 0) throw(...)

I.e. in one IF statement